### PR TITLE
Refactor/CIV-1711-Use optional chaining

### PR DESCRIPTION
**Pull request by laperla11**
**Before creating a pull request make sure that:**

- [X] commit messages are meaningful and follow good commit message guidelines
- [X] README and other documentation has been updated / added (if needed)
- [X] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-1711

### Change description ###
I implemented optional chaining operator for safe access to nested object properties instead of using lodash library. You can find the detail info on the below link :
https://javascript.info/optional-chaining

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/modules/statementOfMeans/dependants/betweenSixteenAndNineteenService.ts [hunk 1]
```diff
@@ -40,13 +38,13 @@ export const saveFormToDraftStore = async (claimId: string, form: BetweenSixteen
 };
 
 const getMaxValue = (claim: Claim): number | undefined => {
-  if (get(claim, 'statementOfMeans.dependants.numberOfChildren')) {
+  if (claim?.statementOfMeans.dependants.numberOfChildren) {
```
[the remark this answers is not in the document]
Comment (TatsyMaslova) [reply src/main/modules/statementOfMeans/dependants/betweenSixteenAndNineteenService.ts:41]: I agree with Felipe. in this case it was checking whether the claim contains the path. hence it was using lodash

File: src/main/routes/features/response/citizenPhoneNumber/citizenPhoneController.ts [hunk 2]
```diff
@@ -21,9 +20,8 @@ function renderView(form: CitizenTelephoneNumber, res: express.Response): void {
 router.get(CITIZEN_PHONE_NUMBER_URL, async (req, res) => {
   try {
     const responseDataRedis: Claim = await getCaseDataFromStore(req.params.id);
-    const citizenTelephoneNumber = !(get(responseDataRedis,'respondent1.telephoneNumber'))
-      ? new CitizenTelephoneNumber()
-      : new CitizenTelephoneNumber(responseDataRedis.respondent1.telephoneNumber);
+    const citizenTelephoneNumber = (responseDataRedis?.respondent1.telephoneNumber)
+      ? new CitizenTelephoneNumber(responseDataRedis.respondent1.telephoneNumber) : new CitizenTelephoneNumber();
```
[the remark this answers is not in the document]
Comment (laperla11) [reply src/main/routes/features/response/citizenPhoneNumber/citizenPhoneController.ts:24]: As the it is checked on line 23, it is not required again here.

File: src/main/modules/statementOfMeans/dependants/betweenSixteenAndNineteenService.ts [hunk 3]
```diff
@@ -40,13 +38,13 @@ export const saveFormToDraftStore = async (claimId: string, form: BetweenSixteen
...
     return Number(claim.statementOfMeans.dependants.numberOfChildren.between16and19);
   }
   return undefined;
 };
 const getNumberOfChildrenLivingWithYou = (claim: Claim): number | undefined => {
-  if (get(claim, 'statementOfMeans.numberOfChildrenLivingWithYou')) {
+  if (claim?.statementOfMeans.numberOfChildrenLivingWithYou) {
```
[the remark this answers is not in the document]
Comment (laperla11) [reply src/main/modules/statementOfMeans/dependants/betweenSixteenAndNineteenService.ts]: made the change.

File: src/main/modules/statementOfMeans/dependants/dependantsService.ts [hunk 4]
```diff
@@ -14,7 +13,7 @@ class DependantsService {
   public async getDependants(claimId: string): Promise<Dependants> {
     try {
       const civilClaimResponse = await getDraftClaimFromStore(claimId);
-      if (get(civilClaimResponse, 'case_data.statementOfMeans.dependants')) {
+      if (civilClaimResponse?.case_data.statementOfMeans.dependants) {
```
[the remark this answers is not in the document]
Comment (laperla11) [reply src/main/modules/statementOfMeans/dependants/dependantsService.ts]: made the change.

File: src/main/modules/statementOfMeans/dependants/dependantsService.ts [hunk 5]
```diff
@@ -27,7 +26,7 @@ class DependantsService {
   public async saveDependants(claimId: string, dependants: Dependants) {
     try {
       const civilClaimResponse = await getDraftClaimFromStore(claimId);
-      if (get(civilClaimResponse, 'case_data.statementOfMeans')) {
+      if (civilClaimResponse?.case_data.statementOfMeans) {
```
[the remark this answers is not in the document]
Comment (laperla11) [reply src/main/modules/statementOfMeans/dependants/dependantsService.ts]: made the change.

File: src/main/routes/features/response/citizenDob/citizenDobController.ts [hunk 6]
```diff
@@ -31,7 +30,7 @@ router.get(DOB_URL, async (req: express.Request, res: express.Response) => {
   try {
     const citizenDob = new CitizenDob();
     const responseDataRedis: Claim = await getCaseDataFromStore(req.params.id);
-    if (get(responseDataRedis,'respondent1.dateOfBirth')) {
+    if (responseDataRedis?.respondent1.dateOfBirth) {
```
[the remark this answers is not in the document]
Comment (laperla11) [reply src/main/routes/features/response/citizenDob/citizenDobController.ts]: made the change.

File: src/main/routes/features/response/citizenPhoneNumber/citizenPhoneController.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (laperla11) [reply src/main/routes/features/response/citizenPhoneNumber/citizenPhoneController.ts:24]: made the change.

File: src/main/routes/features/response/responseType/citizenResponseTypeController.ts [hunk 7]
```diff
@@ -36,7 +36,7 @@ router.get(CITIZEN_RESPONSE_TYPE_URL, async (req, res) => {
   try {
     const citizenResponseType = new CitizenResponseType();
     const responseDataRedis: Claim = await getCaseDataFromStore(req.params.id);
-    if (get(responseDataRedis,'respondent1.responseType')){
+    if (responseDataRedis?.respondent1.responseType){
```
[the remark this answers is not in the document]
Comment (laperla11) [reply src/main/routes/features/response/responseType/citizenResponseTypeController.ts]: made the change.

**Review by GorilaNaranja — APPROVED** [review]

**Review by TatsyMaslova — APPROVED** [review]

**Review by LeonardoPalmeiro — APPROVED** [review]

**Pull request merged**